docs/pages/documentation/11-examples/example-twitter-streaming-public-tweets.md

```diff
@@ -0,0 +1,580 @@
+
+The following function will create a relation between the user and tweet that we specify.
+
+```java
+public static Relation insertUserTweetRelation(GraknGraph graknGraph, Entity user, Entity tweet) {
+  RelationType userTweetRelationType = graknGraph.getRelationType("user-tweet-relation");
+  RoleType postsType = graknGraph.getRoleType("posts");
+  RoleType postedByType = graknGraph.getRoleType("posted_by");
+
+  Relation userTweetRelation = userTweetRelationType.addRelation()
+      .addRolePlayer(postsType, user)
+      .addRolePlayer(postedByType, tweet);
+
+  return userTweetRelation;
+}
+```
+
+### Tweet Insertion Wrap Up
+
+Finally, let's wrap up by defining a function of which the sole responsibility is to execute all of the methods we have defined above.
+
+```java
+public static Relation insertUserTweet(GraknGraph graknGraph, String screenName, String tweet) {
+  Entity tweetEntity = insertTweet(graknGraph, tweet);
+  Entity userEntity = insertUserIfNotExist(graknGraph, screenName);
+  return insertUserTweetRelation(graknGraph, userEntity, tweetEntity);
+}
+```
+
+We're done with tweet insertion functionality! Next step: querying the stored data. Before we proceed, let's add the method we've just defined to the main method as shown below.
+
+```java
+public static void main(String[] args) {
+  try (GraknSession session = Grakn.session(graphImplementation, keyspace)) {
+    withGraknGraph(session, graknGraph -> initTweetOntology(graknGraph)); // initialize ontology
+
+    listenToTwitterStreamAsync(consumerKey, consumerSecret, accessToken, accessTokenSecret, (screenName, tweet) -> {
+      withGraknGraph(session, graknGraph -> insertUserTweet(graknGraph, screenName, tweet)); // insert tweet
+      // TODO: perform some meaningful queries on the inserted data
+    });
+  }
+
+```
+
+## Crafting Simple Queries Using GraQL
+
+We will perform a query which will count the number of tweets a user has posted since the program started. It can be achieved it by utilizing the aggregate query feature. Graql has been chosen over the graph API for this task because it is declarative and therefore much easier to use for complex queries than the graph API.
+
+Let's look at how we can build it step-by-step, start by creating a `QueryBuilder` object which we will use to craft the query in GraQL.
+
+```java
+QueryBuilder qb = graknGraph.graql();
+```
+
+Now let's begin crafting the query. For this tutorial, let's create a `match` query where we retrieve both the `user` and `tweet`.
+
+We will bind them into `var`s which will be named... `user` and `tweet`, respectively. Notice how we deliberately assign the `var`s  identical names as the respective entity types. This is not a necessity and in practice, you are free to name them anything you want.
```

Comment:
Just to add to the intonation, as the names were obvious. It has been removed in 6ccd9fbecbdb34ed673c677d0b1f62a258121687